data-and-functions-2/src/functions/getProductsForOrder.js

```diff
@@ -0,0 +1,18 @@
+import getProductById from './getProductById'
+
+const getProductsForOrder = (data, id) => {
+  if (data == null || id == null || data.orders == null) {
+    throw new Error('Do not have data or ID')
+  }
+  const order = data.orders.find((o) => o.id === id)
+  if (order === undefined) {
```

Comment:
Testing for `undefined` is usually not recomended. It's more of a style thing. I would do one of these tests instead:
```
if (!order) {
```
or
```
if (order == null) {
```